source/_integrations/homekit_controller.markdown

```diff
@@ -79,17 +79,95 @@ HomeKit IP accessories for these device types may work with some caveats:
 
 HomeKit controller will poll your devices, but it will also automatically enable push updates for accessories that support it.
 
-## Bluetooth device support
+## Adding a HomeKit device through Bluetooth
 
-HomeKit Controller will automatically discover Bluetooth devices once the [Bluetooth](/integrations/bluetooth) integration is enabled and functional. Bluetooth devices may take significantly longer to pair than IP devices. Battery-powered devices may require pressing a button on the device to wake it before pairing can be successful.
+You can add a HomeKit device to Home Assistant via [Bluetooth](/integrations/bluetooth).
 
-## Thread device support
+### Prerequisites
 
-In order to use HomeKit over Thread you need a working border router (like a HomePod mini). Your Home Assistant instance will need to be on the same VLAN as the border router.
+- If your Home Assistant instance does not natively support Bluetooth, use an ESPHome Bluetooth proxy.
+  - A proxy can also be helpful if your Home Assistant device is too far away from the device you are trying to pair.
+- If your HomeKit device has been used with Thread before, or is still paired with iOS, reset the device.
+   - HomeKit devices can only be paired to a single controller at once.
+   - If it has been in a Thread network before, the device might remember the Thread credentials of a different network. A reset makes sure the device is not connected to any Thread network.
 
-HomeKit Controller will automatically discover supported Thread devices using the [Zeroconf](/integrations/zeroconf) integration. Battery powered devices may go to sleep and require a button pressing to wake them up before pairing works.
+## To add a HomeKit device through Bluetooth
 
-In order to provision a Thread device onto a mesh network using a HomePod as a border router, first pair it with an iOS device. Then unpair it from the Home app. Don't reset it. This will leave the Thread network details on the device. Home Assistant can now pair with it over Thread.
+1. Power up your HomeKit device.
+   - If you have Bluetooth enabled, the device should be discovered under **{% my integrations title="Settings > Devices & Services" %}**.
+1. Under **{% my integrations title="Settings > Devices & Services" %}**, on the HomeKit integration, select **Configure**.
+     
+     ![HomeKit integration](/images/integrations/homekit_controller/homekit_controller_add_01.png)
+1. To pair the device, enter the HomeKit pairing code. The code is on the device itself or the packaging. 
+   - To pair a battery-powered device, you may need to press a button on the device to wake it.
+   - Bluetooth devices may take significantly longer to pair than IP devices.
+   - Add the device to a room and **Finish**.
+
+## Adding a HomeKit device to a Thread network via Home Assistant
+
+There are two methods to add a HomeKit device to a Thread network:
+
+- via Home Assistant
```

Comment:
"via Home Assistant" on its own is a bit unclear to me. I'd describe it a bit more. The title can still be "via Home Assistant" only.

```suggestion
- via Home Assistant's preferred Thread network
```